### PR TITLE
RSDK-6661, RSDK-6662 Add cloud metadata

### PR DESCRIPTION
**Pull request by maximpertsov**
Adds client- and server-side methods that returns the `robot_part_id`, `primary_org_id`, and `location_id` from a cloud-configured robot.

Blocked by:
* [x] https://github.com/viamrobotics/app/pull/3711
* [x] https://github.com/viamrobotics/api/pull/454

### CODE REVIEW

**Comment by cheukt** [comment]
can you add this to the robot interface and add it to the client as well

File: cloud/cloud.go
Comment (maximpertsov) [inline cloud/cloud.go:1]: created this package to get around circular dependencies. it's probably useful to have a separate package for app-related stuff if end up adding more coupled-features in the future.

File: robot/robot.go [hunk 1]
```diff
@@ -110,6 +111,9 @@ type LocalRobot interface {
 	// ModuleAddress returns the address (path) of the unix socket modules use to contact the parent.
 	ModuleAddress() (string, error)
 
+	// GetCloudMetadata returns app-related information about the robot.
```
Comment (cheukt) [inline robot/robot.go:114]: why not put this into the Robot interface? since the client is implementing it as well

Comment (maximpertsov) [reply robot/robot.go:114]: ugh, because I got lost :disappointed: - meant to put it there, not `LocalRobot`

File: cloud/cloud.go
Comment (maximpertsov) [reply cloud/cloud.go:1]: actually, we have an internal `cloud` package already - will move there instead

**Comment by maximpertsov** [comment]
@benjirewis @cheukt I'm going make this PR a draft again while I add these changes: https://github.com/viamrobotics/rdk/pull/3599#issuecomment-1958095729. Will request reviews again once it's ready

**Comment by maximpertsov** [comment]
> @benjirewis @cheukt I'm going make this PR a draft again while I add these changes: [#3599 (comment)](https://github.com/viamrobotics/rdk/pull/3599#issuecomment-1958095729). Will request reviews again once it's ready

reopened for review

**Review by cheukt — APPROVED** [review]
LGTM

**Pull request merged**